modules/ocf_printhost/files/enforcer

```diff
@@ -0,0 +1,182 @@
+    body=dedent("""\
+        Greetings from the Open Computing Facility,
+
+        This email is letting you know that your most recent print job was
+        rejected since it would exceed your daily quota. The daily quota is
+        {daily_quota} pages today and the semesterly quota is {semester_quota} pages.
+
+        Username: {user}
+        Time: {time}
+        File: {doc_name}
+        Total pages: {pages}
+        Pages left today: {daily_pages}
+        Pages left this semester: {semester_pages}
+
+
+        Does something look wrong? Please reply to
+
+            help@ocf.berkeley.edu
+
+
+        """) + MAIL_SIGNATURE
+)
+
+PRINTER_ERROR_MESSAGE = Message(
+    subject='[OCF] Your latest print job failed',
+    body=dedent("""\
+        Greetings from the Open Computing Facility,
+
+        This email is from the OCF to let you know that your most recent print
+        job failed due to a printer error. If there's something wrong with the
+        printers, please alert the publications staff at the desk.
+
+        Username: {user}
+        Time: {time}
+        File: {doc_name}
+        Total pages: {pages}
+        Pages left today: {daily_pages}
+        Pages left this semester: {semester_pages}
+
+
+        Still can't get it to print? Please reply to
+
+            help@ocf.berkeley.edu
+
+
+        """) + MAIL_SIGNATURE
+)
+
+
+def read_config():
+    conf = ConfigParser()
+    conf.readfp(open(CONF_FILE))
+    mysql_host = conf.get('enforcer', 'user')
+    mysql_passwd = conf.get('enforcer', 'passwd')
+    return mysql_host, mysql_passwd
+
+
+def page_count(path):
+    p = Popen(['/usr/local/bin/enforcer-pc', path], stdout=PIPE)
+    p.wait(30)
+    return int(p.stdout.read())
+
+
+def create_job(env):
+    printer, queue = env['TEAPRINTERNAME'].split('-')
+    return quota.Job(
+        user=env['TEAUSERNAME'],
+        time=datetime.now(),
+        pages=int(env['TEACOPIES']) * page_count(env['TEADATAFILE']),
+        queue=queue,
+        printer=printer,
+        doc_name=env['TEATITLE'],
+        filesize=env['TEAJOBSIZE'],
+    )
+
+
+def send_printer_mail(message, job, quo):
+    body = message.body.format(
+        user=job.user,
+        time=job.time,
+        doc_name=job.doc_name,
+        pages=job.pages,
+        daily_pages=quo.daily,
+        semester_pages=quo.semesterly,
+        daily_quota=quota.daily_quota(),
+        semester_quota=quota.SEMESTERLY_QUOTA,
+    )
+    send_mail_user(job.user, message.subject, body)
+
+
+def prehook(c, job):
+    quo = quota.get_quota(c, job.user)
+    if job.pages > quo.daily:
+        send_printer_mail(INSUFFICIENT_QUOTA_MESSAGE, job, quo)
+        exit(255)
+
+
+def posthook(c, job, success):
+    if success:
+        quota.add_job(c, job)
+    else:
+        quo = quota.get_quota(c, job.user)
+        send_printer_mail(PRINTER_ERROR_MESSAGE, job, quo)
+
+
+def main(argv):
+    try:
+        parser = ArgumentParser(description='Pre- and post-hook for Tea4CUPS')
```

Comment:
Since you wrote a nice docstring, I'm a big fan of doing...

```python
parser = argparse.ArgumentParser(
    description=__doc__,
    formatter_class=argparse.RawDescriptionHelpFormatter,
)
```

then your nice module docstring will be the program's description